koku/api/status/model.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Models to capture server status."""
+
+import logging
+import os
+import platform
+import subprocess
+import sys
+import uuid
+
+from django.db import models
+
+from api import API_VERSION
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Status(models.Model):
+    """A server's status."""
+
+    server_id = models.UUIDField(default=uuid.uuid4,
+                                 editable=False)
+
+    def get_commit(self):  # pylint: disable=R0201
+        """Collect the build number for the server.
+
+        :returns: A build number
+        """
+        commit_info = os.environ.get('KOKU_COMMIT', None)
```

Comment:
@blentz Are we able to "bake" the commit hash into the Openshift build in such a way that its accessible from the app?

Reply:
Good question. There's at least two basic options, I think.

1. We can copy over the .git directory, install git, and let the code execute `git status` or similar.
2. As part of the s2i build, we could drop the commit info into a file, say, in the `.s2i` dir, and just have the code read it from there.

Reply:
After some checking, it looks even easier. If `assemble` updates `.s2i/environment`, those environment variables should be pushed through into the built image's environment variables.